docs/pages/blog/mui-x-v5.md

```diff
@@ -0,0 +1,173 @@
+This was causing an additional re-render with useless data and the X-team had to stay careful to avoid flickering between those fake data and the real ones.
+In v5, the state is initialized synchronously during the first render.
+
+Note that for now the state updates coming from controlled props are still asynchronous.
+If you pass a `props.pageSize`, we will apply it to the state in a `useEffect` and therefore is you read the state just after the render (for instance in a `useLayoutEffect`), you will still see the old version.
+
+## Reduced style specificity for easier customization
+
+In previous versions most of the `DataGrid` and `DataGridPro` components had a CSS specificity of 2, meaning that style overrides and customizations were harder, requiring the developer to look at the DOM tree in order to pick the correct selector.
+With MUI X v5.0.0 we have reduced the CSS specificity of most of the internal `DataGrid` and `DataGridPro` components to 1.
+This will enable developers to more easily change the look and feel of the grid's components.
+
+**Before**
+
+```jsx
+const useStyles = makeStyles(() => ({
+  root: {
+    '& .MuiDataGrid-toolbarContainer': {
+      padding: 50,
+    },
+  },
+}));
+
+const MyCustomToolbar = () => {
+  return <GridToolbarContainer>My custom toolbar</GridToolbarContainer>;
+};
+
+export default function App() {
+  const classes = useStyles();
+
+  return (
+    <div style={{ height: 400, width: '100%' }} className={classes.root}>
+      <DataGrid
+        components={{
+          Toolbar: MyCustomToolbar,
+        }}
+      />
+    </div>
+  );
+}
+```
+
+**After**
+
+```jsx
+const useStyles = makeStyles(() => ({
+  root: {
+    padding: 50,
+  },
+}));
+
+const MyCustomToolbar = () => {
+  const classes = useStyles();
+
+  return (
+    <GridToolbarContainer className={classes.root}>
+      My custom toolbar
+    </GridToolbarContainer>
+  );
+};
+
+export default function App() {
+  return (
+    <div style={{ height: 400, width: '100%' }}>
+      <DataGrid
+        components={{
+          Toolbar: MyCustomToolbar,
+        }}
+      />
+    </div>
+  );
+}
+```
+
+#### Limitations
+
+Although this was a clear improvement we still had to keep the style specificity of 2 for some parts of the `DataGrid` and `DataGridPro`, more specifically the `GridColumnHeaderItem`, `GridRow` and `GridCell` along with all components that are nested in them.
+This is due to performance implications related to how Emotion injects styles into the page.
+To keep the performance of our virtualization engine at its optimal we decided to keep the CSS specificity of 2 for the mentioned components.
+
+## Features highlights
+
+<!-- TODO: add subsections for the features we want to mentune, if any -->
+
+## v4 migration
```

Comment:
@flaviendelangle should we also add part or all of the migration guides you did here?

Reply:
added